pkg/controller/networkpolicy/networkpolicy_controller_test.go

```diff
@@ -15,18 +15,497 @@
+	*NetworkPolicyController
+	podStore                   cache.Store
+	namespaceStore             cache.Store
+	networkPolicyStore         cache.Store
+	appliedToGroupStore        storage.Interface
+	addressGroupStore          storage.Interface
+	internalNetworkPolicyStore storage.Interface
+}
+
+func newController() (*fake.Clientset, *networkPolicyController) {
+	client := newClientset()
+	informerFactory := informers.NewSharedInformerFactory(client, informerDefaultResync)
+	appliedToGroupStore := store.NewAppliedToGroupStore()
+	addressGroupStore := store.NewAddressGroupStore()
+	internalNetworkPolicyStore := store.NewNetworkPolicyStore()
+	npController := NewNetworkPolicyController(client, informerFactory.Core().V1().Pods(), informerFactory.Core().V1().Namespaces(), informerFactory.Networking().V1().NetworkPolicies(), addressGroupStore, appliedToGroupStore, internalNetworkPolicyStore)
+	npController.podListerSynced = alwaysReady
+	npController.namespaceListerSynced = alwaysReady
+	npController.networkPolicyListerSynced = alwaysReady
+	return client, &networkPolicyController{
+		npController,
+		informerFactory.Core().V1().Pods().Informer().GetStore(),
+		informerFactory.Core().V1().Namespaces().Informer().GetStore(),
+		informerFactory.Networking().V1().NetworkPolicies().Informer().GetStore(),
+		appliedToGroupStore,
+		addressGroupStore,
+		internalNetworkPolicyStore,
+	}
+}
+
+func newClientset() *fake.Clientset {
+	client := fake.NewSimpleClientset()
+
+	client.PrependReactor("create", "networkpolicies", k8stesting.ReactionFunc(func(action k8stesting.Action) (bool, runtime.Object, error) {
+		np := action.(k8stesting.CreateAction).GetObject().(*networkingv1.NetworkPolicy)
+
+		if np.ObjectMeta.GenerateName != "" {
+			np.ObjectMeta.Name = fmt.Sprintf("%s-%s", np.ObjectMeta.GenerateName, rand.String(8))
+			np.ObjectMeta.GenerateName = ""
+		}
+
+		return false, np, nil
+	}))
+
+	return client
+}
+
+func TestAddNetworkPolicy(t *testing.T) {
+	ns := metav1.NamespaceDefault
+	npName := "testing-1"
+	pSelector := metav1.LabelSelector{}
+	apgID := getNormalizedUID(generateNormalizedName(ns, &pSelector, nil))
+	npObj := &networkingv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{Name: npName, Namespace: ns},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: pSelector,
+		},
+	}
+	nSelector1 := getK8sLabelSelector(1)
+	nSelectorPeer1 := []networkingv1.NetworkPolicyPeer{
+		{
+			NamespaceSelector: &nSelector1,
+		},
+	}
+	nSelector2 := getK8sLabelSelector(2)
+	nSelectorPeer2 := []networkingv1.NetworkPolicyPeer{
+		{
+			NamespaceSelector: &nSelector2,
+		},
+	}
+	pSelector1 := getK8sLabelSelector(1)
+	pSelectorPeer1 := []networkingv1.NetworkPolicyPeer{
+		{
+			PodSelector: &pSelector1,
+		},
+	}
+	pSelector2 := getK8sLabelSelector(2)
+	pSelectorPeer2 := []networkingv1.NetworkPolicyPeer{
+		{
+			PodSelector: &pSelector2,
+		},
+	}
+	tables := []struct {
+		ir               []networkingv1.NetworkPolicyIngressRule
```

Comment:
Having a name field here can easily point out the error case in a very readable way and save the comment to explain the test case, for example https://github.com/kubernetes/kubernetes/blob/2f58d2e3fe2778ad96fb3e7fb08fdca74b335fe8/cmd/kubelet/app/options/options_test.go#L149
https://github.com/golang/go/wiki/TableDrivenTests#parallel-testing
```
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			...
		})
	}
```

Reply:
thanks for the suggestion.. will update accordingly